pennylane/operation.py

```diff
@@ -561,7 +561,33 @@ def compute_matrix(*params, **hyperparams):  # pylint:disable=unused-argument
         """
         raise MatrixUndefinedError
 
-    def matrix(self, wire_order=None):
+    @property
+    def matrix(self):
+        r"""Matrix representation of an instantiated operator
+        in the computational basis.
+
+        .. warning::
+
+            The ``matrix`` property is deprecated and will be removed in
+            PennyLane version 0.23.
```

Comment:
Should we be less specific here? This may not be a priority to remove so fast?

Reply:
I think this is a good idea!